UPGRADE_GUIDE.md

```diff
@@ -2,9 +2,146 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-csharp` library between major versions.
 
+- [Upgrading from 4.x to 5.0](#upgrading-from-4x-to-50)
 - [Upgrading from 3.x to 4.x](#upgrading-from-3x-to-40)
 - [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
 
+## Upgrading from 4.x to 5.0
+
+### 5.0 High Impact Changes
+
+- [Service Functions](#50-service-functions)
+- [Client Configuration](#50-client-configuration)
+
+### 5.0 Medium Impact Changes
+
+- [Naming Conventions](#50-naming-conventions)
+- [Exception Changes](#50-exception-changes)
+- [Cancellation Tokens](#50-cancellation-tokens)
+- [Parameter Sets](#50-parameter-sets)
+
+### 5.0 Low Impact Changes
+
+- [Drop RestSharp Dependency](#50-drop-restsharp-dependency)
```

Comment:
issue: this is a high impact change, people are no longer beholden to our version of rest sharp and it affects everyone using the package. I'd move it

Reply:
Split the difference, medium